### PR TITLE
#134209515 Give users jwt on successful signup

### PR DESCRIPTION
**Pull request by andela-jare**
#### What does this PR do?
This PR implements json web token at /api/auth/signup endpoints
#### Description of Task to be completed?
Users once authenticated via the signup api should be returned a JWT which to be required on subsequent call to all other endpoints that require authentication
#### How should this be manually tested?
It can be tested using postman rest client
#### Any background context you want to provide?
No background context
#### What are the relevant pivotal tracker stories?
Users should receive a JWT on successful signup
#### Screenshots (if appropriate)
None
#### Questions:
No questions

### CODE REVIEW

**Comment by oagboola** [comment]
fix the houndCi issues

**Comment by oagboola** [comment]
fix all houndCi issues

**Review by makinwab — DISMISSED** [review]
Commit messages not in line with our convention.

Kindly fix.

File: app/controllers/api/auth.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/* eslint amd:true */
+const mongoose = require('mongoose');
+
+const User = mongoose.model('User');
+const jwt = require('jsonwebtoken');
+const config = require('../../../config/env/all');
+//  signup method
+module.exports.signUp = (req, res) => {
```
Comment (makinwab) [inline app/controllers/api/auth.js]: Why the abbreviations?

File: config/env/all.js [hunk 2]
```diff
@@ -5,5 +5,6 @@ const rootPath = path.normalize(path.join(__dirname, '/../..'));
 module.exports = {
   root: rootPath,
   port: process.env.PORT || 3000,
-  db: process.env.MONGOHQ_URL || 'mongodb://localhost:27017/cfh'
+  db: process.env.MONGOHQ_URL || 'mongodb://localhost:27017/cfh',
+  secret: 'kissaSecret'
```
Comment (makinwab) [inline config/env/all.js]: **CRITICAL - SECURITY**
This secret value should be hidden. Make use of an environment variable.

File: test/api/auth.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
+/*  eslint-disable no-unused-vars*/
+/*  eslint amd:true */
+const server = require('../../server');
+
```
Comment (makinwab) [inline test/api/auth.js]: **Nitpick - Cod Quality**
Why the inconsistency with the new lines?

File: test/api/auth.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+const chai = require('chai');
+const mongoose = require('mongoose');
+
+const User = mongoose.model('User');
+
+const chaiHttp = require('chai-http');
+
+const should = chai.should();
+chai.use(chaiHttp);
```
Comment (makinwab) [inline test/api/auth.js:10]: Add a new line here.

**Review by andela-uenabulele — APPROVED** [review]
I have reviewed and think it can be approved

File: app/controllers/api/auth.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+/* eslint amd:true */
+const mongoose = require('mongoose');
+
+const User = mongoose.model('User');
+const jwt = require('jsonwebtoken');
+
+const Auth = {
+  responseData: (res, status, success, message, token) => {
```
Comment (makinwab) [inline app/controllers/api/auth.js]: **Nitpick**
ES6 gives us the option to define objects with methods like this
```
const Auth = {
  responseData(res, status, success, ... ) {
  ...
  },
  
  saveUser() {
  }
}
```

Comment (andela-jare) [reply app/controllers/api/auth.js]: Thanks, I will do that

**Comment by tomily1** [comment]
Good Job, I think it can be approved

File: app/controllers/api/auth.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
+/* eslint amd:true */
+const mongoose = require('mongoose');
+
```
Comment (oagboola) [inline app/controllers/api/auth.js:3]: I'm not sure why you have this empty line here

File: config/routes.js [hunk 7]
```diff
@@ -1,93 +1,99 @@
...
-    app.param('userId', users.user);
+// route.js
+// const async = require('async');
+const users = require('../app/controllers/users');
+
+const answers = require('../app/controllers/answers');
+const questions = require('../app/controllers/questions');
+const avatars = require('../app/controllers/avatars');
+const index = require('../app/controllers/index');
+const jwtAuth = require('../app/controllers/api/auth');
```
[the remark this answers is not in the document]
Comment (oagboola) [reply config/routes.js:7]: check out this

File: app/controllers/api/auth.js (hunk 6 above)
Comment (andela-jare) [reply app/controllers/api/auth.js:3]: Eslint gave this message:
Expected empty line after require statement not followed by another require

**Pull request merged**